tests/test_filer.py

```diff
@@ -108,104 +114,137 @@ def test_upload_file(self, copyMock, copyDirMock):
-                         , stripLines('''
+        self.assertEqual(getTree(dst2),
+                          stripLines('''
                             |-- a
                             |   |-- 1.txt
                             |   `-- 2.txt
                             `-- 3.txt
                             '''
-                            )
-                         )
-
+                                     )
+                          )
 
     def test_getPath(self):
-
-        self.assertEquals( getPath('file:///home/tfga/workspace/cwl-tes/tmphrtip1o8/md5')
-                         ,                '/home/tfga/workspace/cwl-tes/tmphrtip1o8/md5')
-
+        self.assertEqual(
+            getPath('file:///home/tfga/workspace/cwl-tes/tmphrtip1o8/md5'),
+            '/home/tfga/workspace/cwl-tes/tmphrtip1o8/md5')
 
     def test_containerPath(self):
-
-        self.assertEquals( containerPath('/home/tfga/workspace/cwl-tes/tmphrtip1o8/md5')
-                         ,               '/transfer/tmphrtip1o8/md5')
+        self.assertEqual(
+            containerPath('/home/tfga/workspace/cwl-tes/tmphrtip1o8/md5'),
+            '/transfer/tmphrtip1o8/md5')
 
         # What happens if 'path' is not a descendant of HOST_BASE_PATH?
-        self.assertThrows( lambda: containerPath('/someOtherFolder')
-                         , InvalidHostPath
-                         , "'/someOtherFolder' is not a descendant of 'HOST_BASE_PATH' (/home/tfga/workspace/cwl-tes)"
-                         )
-
+        self.assertThrows(lambda: containerPath('/someOtherFolder'),
+                          InvalidHostPath,
+                          "'/someOtherFolder' is not a descendant of "
+                          "'HOST_BASE_PATH' (/home/tfga/workspace/cwl-tes)"
+                          )
 
     def test_newTransput(self):
+        self.assertEqual(newTransput('ftp'), FTPTransput)
+        self.assertEqual(newTransput('http'), HTTPTransput)
+        self.assertEqual(newTransput('https'), HTTPTransput)
+        self.assertEqual(newTransput('file'), FileTransput)
+
+        self.assertThrows(lambda: newTransput('svn'),
+                          UnknownProtocol,
+                          "Unknown protocol: 'svn'"
+                          )
+
+    @patch('ftplib.FTP')
+    def test_ftp_check_directory(self, conn):
+        """ Ensure that when the path provided is an existing directory, the
+            return value is 0."""
+        path = os.path.curdir
+        self.assertEqual(ftp_check_directory(conn, path), 0)
+
+    def test_process_file(self):
+        """ Ensure non-existent scheme returns 1."""
+        filedata = {'url': 'www.foo.bar'}
+        self.assertEqual(process_file('upload', filedata), 1)
+
+    @patch('ftplib.FTP')
+    def test_ftp_make_dirs(self, conn):
+        """ In case of existing directory, exit with 0. """
+        self.assertEqual(ftp_make_dirs(conn, os.curdir), 0)
+
+    def test_file_from_content(self):
```

Comment:
Here you are creating a new file and then deleting it, you could use the fake file system for this.